libfoundation/include/system-dylib.h

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright (C) 2016 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#if !defined(__MCS_SYSTEM_H_INSIDE__)
+#	error "Only <foundation-system.h> can be included directly"
+#endif
+
+/* ================================================================
+ * Errors
+ * ================================================================ */
+
+MC_DLLEXPORT extern MCTypeInfoRef kMCSDylibModuleNotFoundErrorTypeInfo;
+MC_DLLEXPORT extern MCTypeInfoRef kMCSDylibSymbolNotFoundErrorTypeInfo;
+
+/* ================================================================
+ * Dynamic library operations
+ * ================================================================ */
+
+/* Load the module at path. */
+MC_DLLEXPORT extern "C" void *MCSDylibLoadModuleCString(const char *p_path);
```

Comment:
You can safely remove all of the the `extern "C"` qualifiers from this header file; it's always included from within an `extern "C"` block.

I think it might be helpful for explanatory purposes to introduce something like:

```C++
typedef void * MCSDylibModuleRef;
```

And then use it through the API rather than `void *`.

For full type safety in the API, you could even create a distinct type:

```c++
class MCSDylibModule { void *m_handle; };
```